### DISCUSSION TITLE
How to update mme configs?

### DISCUSSION THREAD

**Discussion by happyqiaomai** [magma/magma#5668] [Q&A] [answered]
Hi all,

I have a question about updating mme configs.
I deployed the AGW based on this [post](https://magma.github.io/magma/docs/basics/quick_start_guide).
Accroding to dev notes: /var/opt/magma/configs/gateway.mconfig take precedence over /etc/magma/gateway.mconfig.
I wonder how I can configure mme configs in /var/opt/magma/configs/gateway.mconfig streamed from the cloud. I tried to overwrite it but the configs will get overwritten every 60 seconds. And I didn't find the source of the streamed gateway.mconfig. I tried re-config [magma]/lte/gateway/configs/gateway.mconfig and restart. But it does not work.
In addition, is there any way to force the local gateway to use configs in /etc/magma/gateway.mconfig.

Thanks for your help!
Kelsey

**Comment by amarpad** [comment]
the mconfig gets streamed from the cloud and is generated dynamically, @hcgatewood and @mpgermano probably know best

**Comment by karthiksubraveti** [comment]
@happyqiaomai Can you help understand your usecase better. Why are you trying to overwrite it manually instead of doing this through swagger or NMS ?

**Reply by happyqiaomai** [reply]
Thanks for your reply. I tried with NMS but it does not work. So I was trying other workarounds to modify the configs.

**Comment by hcgatewood** [comment, marked as answer]
Hi @happyqiaomai, we support two mechanisms for changin the MME mconfig

1. Update the values you need via the NMS (the cloud UI), which will then be automatically converted into the relevant MME mconfig values
2. Operate the gateway in headless mode (i.e. disconnected from the Orc8r), and manually input the desired mconfig values

Approach (1) is recommended. Tell us a bit more about what you're trying to do and we can guide you on how to accomplish it via the NMS. Approach (2) is also fine if you're just trying to test out some local dev changes.

If you want to learn more about how mconfigs are generated, you can read our [Orc8r architecture overview](https://magma.github.io/magma/docs/next/orc8r/architecture_overview) and [Orc8r modularity overview](https://magma.github.io/magma/docs/next/orc8r/architecture_modularity)

**Reply by happyqiaomai** [reply]
Thanks for the links! I'll try with the local mode. What's the proper way to enforce the local mode? I have been modifying source codes to bypass the config updates, which must not be the recommended way.

**Reply by hcgatewood** [reply]
Should be sufficient to clear the contents (or values) of `control_proxy.yml` https://magma.github.io/magma/docs/next/lte/deploy_config_agw#access-gateway-configuration

cc @sudhikan as he probably knows if there's a better way

**Reply by sudhikan** [reply]
Best way to make sure we aren't pulling config is to turn off `config_streaming` in `/etc/magma/magmad/yaml` and restart magmad. Does that help?
